### PR TITLE
add course info methods

### PR DESCRIPTION
**Pull request by ksimuk**

### CODE REVIEW

File: src/lib/course.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
+import bent from 'bent'
+import config from './config'
+
+const getJson = bent('json')
+
+export type Assignement = {
```
Comment (vvlevykin) [inline src/lib/course.ts:6]: typo

Comment (ksimuk) [reply src/lib/course.ts]: fixed

File: src/lib/course.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    const token = config.getToken()
+    const domain = config.getDomain()
+    const authHeaders = {
+      'Authorization': `Bearer ${token}`
+    }
+    const res = await getJson(`https://octopus.${domain}/api/v1/courses/${courseId}/assignments/${assignmentId}/students`, undefined, authHeaders)
+    for (const progress of res) {
+      if (progress.completion_date) {
+        const t = new Date(1970, 0, 1); // Epoch
+        t.setSeconds(progress.completion_date.seconds);
```
Comment (vvlevykin) [inline src/lib/course.ts:62]: maybe `setUTCSeconds`? Also, maybe it's better to extract this to a helper function.

File: src/lib/course.ts [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+      throw new Error(message)
+    }
+    throw error
+  }
+}
+
+function fixDate(progress: StudentProgress) {
+  if (progress.completion_date) {
+    const t = new Date(1970, 0, 1); // Epoch
+    t.setUTCSeconds(progress.completion_date.seconds);
```
Comment (vvlevykin) [inline src/lib/course.ts:53]: Maybe it's better to extract this to a helper function.

**Review by vvlevykin — DISMISSED** [review]

File: src/lib/course.ts (hunk 3 above)
Comment (ksimuk) [reply src/lib/course.ts]: fixed

**Review by vvlevykin — APPROVED** [review]

**Pull request merged**